lib/evm_database_ops.rb

```diff
@@ -119,15 +118,19 @@ def self.restore(db_opts, connect_opts = {})
         backup_folder = action == :dump ? "db_dump" : "db_backup"
         uri = File.join(connect_opts[:uri], backup_folder, connect_opts[:remote_file_name])
       end
+    else
+      uri = db_opts[:local_file]
+    end
 
-      session = MiqGenericMountSession.new_session(connect_opts)
-      db_opts[:local_file] = session.uri_to_local_path(uri)
+    MiqFileStorage.with_interface_class(connect_opts) do |file_storage|
+      send_args = [uri, db_opts[:byte_count]].compact
+      file_storage.send(STORAGE_ACTIONS_TO_METHODS[action], *send_args) do |input_path|
+        db_opts[:local_file] = input_path
+        yield(db_opts)
+      end
     end
 
-    block_result = yield(db_opts, session, uri) if block_given?
-    uri || block_result
```

Comment:
Do you know why we might return the `block_result` here? I can't remember, but it looks like a slight change in behavior.

Reply:
I would have to look at what I have done when messing with this method over the past months, but I was pretty sure I was trying to keep it the same.

Will look and report back.

Reply:
Okay, there is a bit of a :book: with this one, but it is a very minor change in behavior, but I am pretty sure the result is the same regardless (and we don't actually do anything with the result, for what it is worth... at least in the rake tasks).

* * *

So this first got added here:  https://github.com/ManageIQ/manageiq/pull/17483

Which effectively was a refactoring of the of all of backup/restore methods to use the same mounting code, since that was the same for both operations.  The `block_result` was only returned for `:local_file` based actions, because the `uri` wouldn't be set.  This would just return the file that was generated by `PostgresAdmin`, which was basically `:local_file` anyway.

Since everything is flowing through a subclass of `MiqFileStorage` now, everything requires that `uri` is set, even though now it might just be a `:local_file`.  Hence why saving the `block_result` to a local variable is no longer needed.

Hope that makes sense.